### DISCUSSION TITLE
What are the limits for HostIds under Invoke-FalconRtr

### DISCUSSION THREAD

**Discussion by bryaner** [CrowdStrike/psfalcon#271] [Q&A] [answered]
I find the below Powershell scrpit (remove_SOFTWARE) do not apply to all the $Host where it has more than 100 records.
Do you know what is a better way to run a uploaded powershell scripts to a large number of hosts. What is the limit of -HostIds

**$Hosts = Get-FalconHost -Filter "platform_name:'Linux'"

Invoke-FalconRtr -Command runscript -Arguments "-CloudFile='remove_SOFTWARE'" -HostIds $Hosts -Timeout 600 -QueueOffline $true**

**Comment by bk-cs** [comment, marked as answer]
Your problem is the first line:
```
$Hosts = Get-FalconHost -Filter "platform_name:'Linux'"
```
This will return 100 results by default. Add the `-All` switch to return all results.

`Invoke-FalconRtr` does not have a hosts limit. It will generate sessions of 10,000 hosts and loop through them, no matter how many there are.

**Reply by bryaner** [reply]
Thank you very much and pardon me for such a silly question.

I tried to trigger a PS Script which removes a software using MSIEXEC (which take about 1 min to execute) but it doesn't seems to work on all machines. However if I do a loop, its more reliable.
Do you think the problem is the PS Script (uninstallation) or the way how RTR works, it timeout waiting for the response.

**$IDS = Get-FalconHost -Filter "tags:'FalconGroupingTags/Batch1'"
$i=0
foreach ($data in $IDS)
{
	$i++ 
	Invoke-FalconRtr -Command runscript -Arguments "-CloudFile='remove_SOFTWARE'" -HostIds $data -Timeout 600 -QueueOffline $true
}**




********PS SCRIPT TO UNINSTALL A SOFWARE***************
    Start-Process "C:\Windows\System32\msiexec.exe" `
        -ArgumentList "/x {ABC833A1-AF24-4CAE-82DF-CFE14C14BXYZ} /qn /norestart" -Wait

**Reply by bk-cs** [reply]
Real-time Response will time out if the session is not kept alive. `Invoke-FalconRtr` is not designed to keep the session alive (but I am working on adding this in the next release). 

To avoid the timeout issue, you can use a secondary process instead. The easiest way is to modify your last line:
```powershell
Start-Process "C:\Windows\System32\msiexec.exe" -ArgumentList "/x {ABC833A1-AF24-4CAE-82DF-CFE14C14BXYZ} /qn /norestart" -PassThru | ForEach-Object { "Started $($_.Name) [$($_.Id)]" }
```

**Reply by bryaner** [reply]
Thank you for your answer, greatly appreciated.
Will give it a try next week.
